catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/changeProjectOrderTest.java

```diff
@@ -0,0 +1,170 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.espresso.Espresso.closeSoftKeyboard;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.Espresso.pressBack;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.RootMatchers.isDialog;
+import static android.support.test.espresso.matcher.ViewMatchers.hasDescendant;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.common.FlavoredConstants.DEFAULT_ROOT_DIRECTORY;
+import static org.catrobat.catroid.uiespresso.ui.fragment.rvutils.RecyclerViewInteractionWrapper.onRecyclerView;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.fail;
+
+@RunWith(AndroidJUnit4.class)
+public class changeProjectOrderTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<ProjectListActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ProjectListActivity.class, true, false);
+
+	private String ProjectOneName = "ProjectOneName";
+	private String ProjectTwoName = "ProjectTwoName";
+
+	@Before
+	public void setUp() throws Exception {
+		createProject(ProjectOneName);
+		createProject(ProjectTwoName);
+
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void changeProjectOrderTest() {
+		try{
+			Thread.sleep(2000);
+		}
+		catch (Exception e){}
+		onRecyclerView().performOnItemWithText(ProjectTwoName,click());
+		pressBack();
+
+		List<ProjectData> items = new ArrayList<>();
+		for (String projectName : FileMetaDataExtractor.getProjectNames(DEFAULT_ROOT_DIRECTORY)) {
+			File codeFile = new File(PathBuilder.buildPath(PathBuilder.buildProjectPath(projectName), Constants.CODE_XML_FILE_NAME));
+			items.add(new ProjectData(projectName, codeFile.lastModified()));
+		}
+
+		Collections.sort(items, new Comparator<ProjectData>() {
+			@Override
+			public int compare(ProjectData project1, ProjectData project2) {
+				return Long.valueOf(project2.lastUsed).compareTo(project1.lastUsed);
+			}
+		});
+
+		if(!items.get(0).projectName.equals(ProjectTwoName))
+		{
+			fail("order not updated");
```

Comment:
no.
no if -> fail constructs, this is what assert is for, in this case:
assertEquals(ProjectTwoName, items.get(0).projectName); will replace the if block.